### PR TITLE
FEM: fixes

### PR DESCRIPTION
**Pull request by berndhahnebach**
- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)

Only changes in FEM code.

This fixes one of the biggest limitation of FEM. The analysis of beam element if the geometry is in z-axis direction was never possible. This adds a beam rotation object and makes it possible to analyze beams in z-axis. 

Furthermore lots of small fixes and the DataAtPoint for vtk.

From my point of view after this PR I start to only fix bugs for FEM. There a  few of them around. 

cheers bernd

https://forum.freecadweb.org/viewtopic.php?f=18&t=24878
https://forum.freecadweb.org/viewtopic.php?f=18&t=20801

### CODE REVIEW

File: src/Mod/Fem/femmesh/meshtools.py [hunk 1]
```diff
@@ -401,15 +514,17 @@ def get_femelement_sets_from_group_data(femmesh, fem_objects):
...
+        return 'R' + str(i)
     elif hasattr(obj, "Proxy") and obj.Proxy.Type == 'Fem::FemElementFluid1D':
-        return 'Fluid' + str(i)
+        return 'F' + str(i)
     elif hasattr(obj, "Proxy") and obj.Proxy.Type == 'Fem::FemElementGeometry2D':
-        return 'Shell' + str(i)
+        return 'S' + str(i)
     else:
-        print('Error: ', obj.Name, ' --> ', obj.Proxy.Type)
+        FreeCAD.Console.PrintError('Error in creating short elset name for obj: ' + obj.Name + ' --> Proxy.Type: ' + str(obj.Proxy.Type) + '\n')
```
Comment (luzpaz) [inline src/Mod/Fem/femmesh/meshtools.py]: `elset`?

File: src/Mod/Fem/femtools/ccxtools.py [hunk 2]
```diff
@@ -410,10 +418,10 @@ def check_prerequisites(self):
         if self.beam_sections:
             if self.shell_thicknesses:
                 # this needs to be checked only once either here or in shell_thicknesses
-                message += "Beam Sections and shell thicknesses in one analysis is not supported at the moment.\n"
+                message += "Beam sections and shell thicknesses in one analysis is not supported at the moment.\n"
             if self.fluid_sections:
                 # this needs to be checked only once either here or in shell_thicknesses
-                message += "Beam Sections and Fluid Sections in one analysis is not supported at the moment.\n"
+                message += "Beam sections and fluid Sections in one analysis is not supported at the moment.\n"
```
Comment (luzpaz) [inline src/Mod/Fem/femtools/ccxtools.py]: 2nd `Sections` should be lowercase

File: src/Mod/Fem/femsolver/calculix/writer.py [hunk 3]
```diff
@@ -337,14 +337,18 @@ def write_element_sets_material_and_femelement_type(self, f):
         f.write('** Element sets for materials and FEM element type (solid, shell, beam, fluid)\n')
         f.write('** written by {} function\n'.format(sys._getframe().f_code.co_name))
 
+        # in any case if we have beams, we gone need the element ids for the rotation elsets
```
Comment (luzpaz) [inline src/Mod/Fem/femsolver/calculix/writer.py]: `we gone need` should be either `we're going to need` or `we're gonna need`

**Comment by luzpaz** [comment]
@berndhahnebach please see https://github.com/berndhahnebach/FreeCAD_bhb/pull/28

**Comment by berndhahnebach** [comment]
added luzpaz typo commit

**Comment by wwmayer** [comment]
Merged.

**Pull request closed without merging**